FAQ Entry for Singularity Bind Path Issues (#131)
nf-core/nf-co.re: markdown/usage/troubleshooting.md

    INFO:    Creating SIF file...
    FATAL:   Unable to pull docker://nfcore/rnaseq:1.3: While searching for mksquashfs: exec: "mksquashfs": executable file not found in $PATH
```

> In such a case, please install `mksquashfs` or ask in your IT department that this package is installed for you.

Reviewer: ```suggestion
If this is the case, please install `mksquashfs` or ask your IT department to install the package for you.
```

## Cannot find input files when using Singularity

> Depending on how you install Singularity on your system, there are several often occuring caveats. Typically these result in error messages such as these:

Reviewer: ```suggestion
Depending on how you install Singularity on your system several reoccurring issues have been reported. Typically these result in error messages such as these:
```

```bash

Command error:
  �[91mERROR  : Failed to resolve path to /home/bla/singularity/mnt/container: No such file or directory
  �[0m�[31mABORT  : Retval = 255
  �[0m
```

> You can check to verify that this is indeed a problem by using a Singularity Shell into your container and checking whether your required paths are available **inside** your container:

Reviewer: ```suggestion
You can verify if this is indeed a problem by using a Singularity Shell to access your container, and to check whether your required paths are available **inside** your container:
```

```bash
singularity shell /path/to/work/singularity/imagename.simg
ls -la /path/to/work
ls -la /path/to/network_storage
```

> If any of these `ls -la` returns a `Not Found` error, you will need to get Singularity Bind Paths set/updated on your system.

Reviewer: ```suggestion
If any of these `ls -la` commands returns a `Not Found` error, you will need to set/update Singularity Bind Paths on your system.
```

> The Singularity installation requires certain host paths to be bound. Check out [Singularity Bind Paths Documentation](https://sylabs.io/guides/3.0/user-guide/bind_paths_and_mounts.html) for more information on this in detail. In many cases this can be resolved by adding these paths to your `/etc/singularity/singularity.conf` as illustrated on the document linked above:

Reviewer: ```suggestion
The Singularity installation requires certain host paths to be bound. Please see [Singularity Bind Paths Documentation](https://sylabs.io/guides/3.0/user-guide/bind_paths_and_mounts.html) for a more detailed explanation. In many cases this can be resolved by adding these paths to your `/etc/singularity/singularity.conf` as highlighted in the documentation:
```

```bash
bind path = /beegfs/work/
bind path = /scratch
bind path = /gpfs
bind path = /home
```

> Optionally, you can also add Singularity Bind Paths to your Nextflow call, e.g. using `autoMounts` and/or `runOptions` in the [Singularity scope](https://www.nextflow.io/docs/latest/config.html#config-singularity)

Reviewer: ```suggestion
Alternatively, you can also add Singularity Bind Paths to your Nextflow call, e.g. using `autoMounts` and/or `runOptions` in the [Singularity scope](https://www.nextflow.io/docs/latest/config.html#config-singularity)
```
